src/vumi2/transports/smpp/smpp.py

```diff
@@ -57,18 +57,28 @@ async def setup(self) -> None:
         self.stream = await open_tcp_stream(
             host=self.config.host, port=self.config.port
         )
+        send_channel, receive_channel = open_memory_channel(0)
         self.client = EsmeClient(
             self.nursery,
             self.stream,
             self.config,
             self.sequencer,
             self.submit_sm_processor,
+            send_channel,
         )
         await self.client.start()
         self.connector = await self.setup_receive_outbound_connector(
             connector_name=self.config.transport_name,
             outbound_handler=self.handle_outbound,
         )
+        self.nursery.start_soon(self.handle_inbound_message_or_event, receive_channel)
+
+    async def handle_inbound_message_or_event(
+        self, receive_message_channel: MemoryReceiveChannel
+    ) -> None:
+        async for msg in receive_message_channel:
+            if isinstance(msg, Event):
+                await self.connector.publish_event(msg)
```

Comment:
Do we want to log non-event messages here? Presumably when we add support for inbound messages we could change that to a `cast` or something.

Reply:
Yeah so if we receive anything other than what we expect here, since this is us sending it internally, it means it's a bug in the code. So I'll add a error level log here.